analyses/molecular-subtyping-LGAT/03-subset-cnv-files-LGAT.Rmd

```diff
@@ -0,0 +1,329 @@
+
+### Setup
+```{r}
+library("tidyverse")
+```
+
+### Input
+```{r}
+# Look for git root folder
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+
+# get subset folder
+subset_dir <- file.path(
+  root_dir,
+  "analyses",
+  "molecular-subtyping-LGAT",
+  "lgat-subset"
+)
+
+# create if doesn't exist
+if (!dir.exists(subset_dir)) {
+  dir.create(subset_dir)
+}
+
+# manta calls
+manta <- read_tsv(file.path(
+  root_dir,
+  "data",
+  "pbta-sv-manta.tsv.gz"
+))
+
+# consensus seg calls
+cnv <- read_tsv(file.path(
+  root_dir,
+  "data",
+  "pbta-cnv-consensus.seg.gz"
+))
+
+# consensus seg annotated with gene
+cnv_anno <- read_tsv(file.path(
+  root_dir,
+  "data",
+  "consensus_seg_annotated_cn_autosomes.tsv.gz"
+))
+
+# Gene location and domain overlap file
+# Input files used for pfam info (hg38)
+# http://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/pfamDesc.txt.gz
+# http://hgdownload.soe.ucsc.edu/goldenPath/hg38/database/ucscGenePfam.txt.gz
+# biomaRt gene/pfam locations call (hg38)
+# hsapiens_gene_ensembl == Human genes (GRCh38.p13) 10th Feb 2020
+# ensembl <- biomaRt::useMart(biomart = "ENSEMBL_MART_ENSEMBL",dataset = "hsapiens_gene_ensembl",host="ensembl.org")
+# Code used to generate pfamDataBioMart.RDS in https://gist.github.com/kgaonkar6/02b3fbcfeeddfa282a1cdf4803704794
+
+bioMartDataPfamFGFR1 <-
+  readRDS(system.file("extdata", "pfamDataBioMart.RDS", package = "annoFuse")) %>%
+  # to check if biospecimen harbors FGFR1 TKD (tyrosine kinase domain tandem duplication)
+  dplyr::filter(
+    hgnc_symbol == "FGFR1",
+    DESC == "Protein tyrosine kinase"
+  )
+```
+
+### Gather lgat subset
+
+```{r}
+# File from 00-LGAT-select-pathology-dx that is used for the pathology diagnosis
+# was used to subset clinical file
+lgat_specimens_df <- read_tsv(file.path(subset_dir, "lgat_metadata.tsv"))
+
+# Filter to RNA-Seq samples
+lgat_wgs_biospecimen_ids <- lgat_specimens_df %>%
+  dplyr::filter(experimental_strategy == "WGS") %>%
+  dplyr::pull(Kids_First_Biospecimen_ID)
+
+# Filter CNV calls
+manta_lgat <- manta %>%
+  dplyr::filter(Kids.First.Biospecimen.ID.Tumor %in% lgat_wgs_biospecimen_ids)
+
+cnv_lgat <- cnv %>%
+  dplyr::filter(ID %in% lgat_wgs_biospecimen_ids)
+```
+
+### Overlap tandem duplication in FGFR1 with exons 10-18 which make up the kinase domain, per [comment](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/790#issuecomment-729229244).
+
+First, create genomic ranges objects for CNV seg file and gene.
+
+```{r}
+# Genomic range for cnv seg file
+cnv_gr <- cnv %>%
+  # format
+  dplyr::rename(
+    chr = chrom, start = loc.start, end = loc.end,
+    copy_number = copy.num
+  ) %>%
+  dplyr::select(-num.mark, -seg.mean) %>%
+  # make genomic ranges
+  GenomicRanges::makeGRangesFromDataFrame(
+    keep.extra.columns = TRUE,
+    starts.in.df.are.0based = FALSE
+  )
+
+# Genomic range for gene location and domain overlap file
+domain_gr <- bioMartDataPfamFGFR1 %>%
```

Comment:
So this range only contains the Protein tyrosine kinase exon, is that correct?

Reply:
Correct, I filter to keep only protein tyrosine kinase *domain* region.